OpenRA.Mods.Common/Pathfinder/PathSearch.cs

```diff
@@ -116,6 +116,38 @@ public override CPos Expand()
 				// Calculate the cost up to that point
 				var gCost = currentCell.CostSoFar + connection.Cost;
 
+				// This constant is experimental. Orientation.Yaw is integer from [0...1024]
+				// 128 is eight of it => difference between 2 neighboring orientations
+				// In future we should check how long it takes to rotate on same position.
+				// longer it takes, smaller variable a
+				var a = 128.0;
+
+				// Check orientation of actor/vehicle and make directions in front of it more interesting among all eight possible
+				if (connection.Destination.X - currentMinNode.X == 1)
+				{
+					if (connection.Destination.Y - currentMinNode.Y == 1)
+						gCost += (int)Math.Round(Math.Abs(512 - Math.Abs(orientation.Yaw.Angle - 896)) / a);
+					else if (connection.Destination.Y - currentMinNode.Y == 0)
+						gCost += (int)Math.Round(Math.Abs(512 - Math.Abs(orientation.Yaw.Angle - 768)) / a);
+					else
+						gCost += (int)Math.Round(Math.Abs(512 - Math.Abs(orientation.Yaw.Angle - 640)) / a);
+				}
+				else if (connection.Destination.X - currentMinNode.X == -1)
+				{
+					if (connection.Destination.Y - currentMinNode.Y == 1)
+						gCost += (int)Math.Round(Math.Abs(512 - Math.Abs(orientation.Yaw.Angle - 128)) / a);
+					else if (connection.Destination.Y - currentMinNode.Y == 0)
+						gCost += (int)Math.Round(Math.Abs(512 - Math.Abs(orientation.Yaw.Angle - 256)) / a);
+					else
+						gCost += (int)Math.Round(Math.Abs(512 - Math.Abs(orientation.Yaw.Angle - 384)) / a);
+				}
+				else if (connection.Destination.X - currentMinNode.X == 0)
+				{
+					if (connection.Destination.Y - currentMinNode.Y == -1)
+						gCost += (int)Math.Round(Math.Abs(512 - Math.Abs(orientation.Yaw.Angle - 512)) / a);
+					else
+						gCost += (int)Math.Round(Math.Abs(512 - Math.Abs(orientation.Yaw.Angle)) / a);
+				}
```

Comment:
Aren't we just changing a single value? I.e. something like
```
var value = 0;
if ...
    value = xy;
else ...
    value = z;

gCost += (int)Math.Round(Math.Abs(512 - Math.Abs(orientation.Yaw.Angle - value)) / a);
```
would be better imo.

Reply:
yup. me dummy